### PR TITLE
avm2: Add SWFv21 and above XML special case

### PR DESCRIPTION
**Pull request by sleepycatcoding**
Fixes SWF v21 and above XML appendChild test.

### CODE REVIEW

File: core/src/avm2/e4x.rs [hunk 1]
```diff
@@ -1249,17 +1267,33 @@ pub fn maybe_escape_child<'gc>(
...
                 .construct(activation, &[string.into()])?;
-            Ok(xml.into())
+            return Ok(xml.into());
         }
-    } else {
-        Ok(child)
     }
+
+    // FIXME: This should use caller movie version.
+    if activation.context.swf.version() >= 21 {
```
Comment (Lord-McSweeney) [inline core/src/avm2/e4x.rs:1283]: `activation.caller_movie().unwrap().version()`

Comment (sleepycatcoding) [reply core/src/avm2/e4x.rs:1283]: Originally used that, but some tests were panicking since caller_movie was missing?

Comment (Lord-McSweeney) [reply core/src/avm2/e4x.rs:1283]: I'm PRing a fix right now

Comment (Lord-McSweeney) [reply core/src/avm2/e4x.rs:1283]: Tests should no longer fail if you rebase

**Comment by Lord-McSweeney** [comment]
Can you add a test that ensures that it's really the current SWF's version that's being used?

**Comment by sleepycatcoding** [comment]
> Can you add a test that ensures that it's really the current SWF's version that's being used?

Added a test, I hope that it is sufficient. It uses `flash.display.Loader` to load a version 21 SWF, while the root is version 20.

**Review by Lord-McSweeney — APPROVED** [review]

**Review by Dinnerbone — APPROVED** [review]

**Pull request merged**